### PR TITLE
Remove tinyxml2 dependency. Implement basic xml parser just sufficien…

### PR DESCRIPTION
**Pull request by gmcode**
…t to parse mappings file.

### CODE REVIEW

File: tz.cpp [hunk 1]
```diff
@@ -352,36 +345,164 @@ get_download_mapping_file(const std::string& version)
...
 static
-std::vector<date::detail::timezone_mapping>
-load_zone_mappings_from_xml_file(const std::string& input_path)
+std::vector<detail::timezone_mapping>
+load_timezone_mappings_from_xml_file(const std::string& input_path)
 {
-    using tinyxml2::XMLDocument;
-    std::vector<date::detail::timezone_mapping> zone_map_list;
-    XMLDocument doc;
+    size_t line_num = 0;
```
Comment (HowardHinnant) [inline tz.cpp]: `std::size_t`

File: tz.cpp [hunk 2]
```diff
@@ -352,36 +345,164 @@ get_download_mapping_file(const std::string& version)
...
         msg += input_path;
-        msg += '\"';
+        msg += "\".";
         throw std::runtime_error(msg);
     }
 
-    auto supplementalData = doc.FirstChildElement("supplementalData");
-    auto windowsZones = supplementalData->FirstChildElement("windowsZones");
-    auto mapTimeZones = windowsZones->FirstChildElement("mapTimezones");
+    auto error = [&](const char* info)
```
Comment (HowardHinnant) [inline tz.cpp]: I would prefer explicit captures:
````
 auto error = [&input_path, &line_num](const char* info)
````
Rationale:  Effective Modern C++ Item 31is pretty convincing.

File: tz.cpp [hunk 3]
```diff
@@ -352,36 +345,164 @@ get_download_mapping_file(const std::string& version)
...
+        std::string msg = "Error loading time zone mapping file \"";
+        msg += input_path;
+        msg += "\" at line ";
+        msg += std::to_string(line_num);
+        msg += ": ";
+        msg += info;
+        throw std::runtime_error(msg);
+    };
+    // [optional space]a="b"
+    auto read_attribute = [&](const char* name, std::string& value, size_t startPos)->size_t
```
Comment (HowardHinnant) [inline tz.cpp]: Ditto on the explicit captures here.

File: tz.cpp [hunk 4]
```diff
@@ -352,36 +345,164 @@ get_download_mapping_file(const std::string& version)
...
+    // Quick but not overly forgiving XML mapping file processing.
+    bool mapTimezonesOpenTagFound = false;
+    bool mapTimezonesCloseTagFound = false;
+    bool mapZoneOpenTagFound = false;
+    bool mapTZoneCloseTagFound = false;
+    size_t mapZonePos = std::string::npos;
+    size_t mapTimezonesPos = std::string::npos;
+    static char mapTimeZonesOpeningTag[] = "<mapTimezones ";
+    static char mapZoneOpeningTag[] = "<mapZone ";
+    size_t mapZoneOpeningTagLen = sizeof(mapZoneOpeningTag) / sizeof(mapZoneOpeningTag[0]) - 1;
```
Comment (HowardHinnant) [inline tz.cpp]: I think `CONSTDATA` would be a good qualifier for the above three variables, and then you could drop the `static` on the first two.
````
CONSTDATA char mapTimeZonesOpeningTag[] = "<mapTimezones ";
// etc.
````

**Comment by HowardHinnant** [comment]
Overall looks awesome, thanks!

**Comment by gmcode** [comment]
Hi Howard, all your changes are addressed in a new pull #65.
I started the git process again, wrestling with git is a job in of itself. Sorry if this causes anyone any trouble.

**Pull request closed without merging**